client/lib/plugins/utils.js

```diff
@@ -243,3 +244,37 @@ export function normalizePluginsList( pluginsList ) {
 export function filterNotices( logs, siteId, pluginId ) {
 	return filter( logs, filterNoticesBy.bind( this, siteId, pluginId ) );
 }
+
+/**
+ * Regex to extract the author from the search
+ */
+export const AUTHOR_PATTERN = /author:(?:\s)*"(.*)"/;
+
+/**
+ * Extract author and search params from the plugin search query
+ *
+ * @param {string} searchTerm The full plugin search query
+ * @returns {Array<string|null>} The first item will be the search and the second will be the author if exists
+ */
+export function extractSearchInformation( searchTerm = '' ) {
+	const author = searchTerm.match( AUTHOR_PATTERN )?.[ 1 ] || null;
+	const search = searchTerm.replace( AUTHOR_PATTERN, '' ).trim();
+
+	return [ search, author ];
+}
+
+export const WPORG_PROFILE_URL = 'https://profiles.wordpress.org/';
+
+/**
+ * Returns an author keyword to be used on plugin search by author
+ *
+ * @param plugin
+ * @returns {string} the keyword
+ */
+export function getPluginAuthorKeyword( plugin ) {
+	if ( plugin?.author_profile?.includes( WPORG_PROFILE_URL ) ) {
+		return plugin.author_profile.replace( WPORG_PROFILE_URL, '' ).replaceAll( '/', '' );
+	}
```

Comment:
We can use a regex pattern here eg

`'https://profiles.wordpress.org/joostdevalk/'.match(/https:\/\/profiles.wordpress.org\/(.*)\//)`

will return an array with `[1] : "joostdevalk"`

Reply:
Yes, I thought about regex at first, but I decided to go with `replaceAll` cause regex has some performance problems in some edge cases. 

Also, I usually think of regex when we have dynamic values, as we would always have the same string I decided to go with a fixed string.